sympy/sets/tests/test_fancysets.py

```diff
@@ -656,9 +657,7 @@ def test_infinitely_indexed_set_2():
 def test_imageset_intersect_real():
     from sympy import I
     from sympy.abc import n
-    assert imageset(Lambda(n, n + (n - 1)*(n + 1)*I), S.Integers).intersect(S.Reals) == \
-            FiniteSet(-1, 1)
-
+    assert imageset(Lambda(n, n + (n - 1)*(n + 1)*I), S.Integers).intersect(S.Reals) == Complement(S.Integers, FiniteSet((-1, 1)))
```

Comment:
This doesn't seem correct

Reply:
@smichr @shashankks0987 this output is incorrect:
```python
In [8]: S1 = imageset(Lambda(n, n + (n - 1)*(n + 1)*I), S.Integers)

In [9]: S1
Out[9]: {n + ⅈ⋅(n - 1)⋅(n + 1) │ n ∊ ℤ}

In [10]: 2 in S1
Out[10]: False

In [11]: 2 in S1.intersect(Reals)
Out[11]: True
```
What is actually being returned is:
```python
In [12]: S1.intersect(S2)
Out[12]: ℤ \ {(-1, 1)}
```
That's the set of integers minus the set containing a single tuple `(-1, 1)`. Something has clearly gone wrong there. The previous answer was correct. I think it's fine if this intersection doesn't evaluate but it shouldn't return a wrong result like this.

Reply:
Ok I'll try fixing this

Reply:
```
In [4]: S1
Out[4]: {n + ⅈ⋅(n - 1)⋅(n + 1) │ n ∊ ℤ}

In [5]: 2 in S1
Out[5]: False

In [6]: 2 in S1.intersect(Reals)
Out[6]: False

In [7]: S2 = Reals

In [8]: S1.intersect(S2)
Out[8]: {-1, 1}
```
This seems to be working fine though

Reply:
With the new diff @oscarbenjamin